server/src/main/java/org/opensearch/extensions/ExtensionRequest.java

```diff
@@ -25,20 +25,29 @@
 public class ExtensionRequest extends TransportRequest {
     private static final Logger logger = LogManager.getLogger(ExtensionRequest.class);
     private ExtensionsOrchestrator.RequestType requestType;
+    private String failureException;
 
     public ExtensionRequest(ExtensionsOrchestrator.RequestType requestType) {
         this.requestType = requestType;
+        this.failureException = "";
     }
 
     public ExtensionRequest(StreamInput in) throws IOException {
         super(in);
         this.requestType = in.readEnum(ExtensionsOrchestrator.RequestType.class);
+        this.failureException = in.readOptionalString();
+    }
+
+    public ExtensionRequest(ExtensionsOrchestrator.RequestType requestType, String failureException) {
+        this.requestType = requestType;
+        this.failureException = failureException;
```

Comment:
Should we change the ` ExtensionRequest` here with `failureException` because `ExtensionRequest` was created to be generic for all the requests? @saratvemulapalli any thoughts here will be appreciated. Thanks

Reply:
I'll address this in a follow-up PR

Reply:
@ryanbogan do create a separate issue for tracking the follow-up PR so that we dont miss it and we can unblock this PR from merging. Thanks

Reply:
https://github.com/opensearch-project/opensearch-sdk-java/issues/135